src/Microsoft.SqlTools.ServiceLayer/TableDesigner/Contracts/Components/TableProperties.cs

```diff
@@ -23,11 +32,42 @@ public class TableProperties<T> : ComponentPropertiesBase
         /// <summary>
         /// All properties of the object.
         /// </summary>
-        public DesignerDataPropertyInfo[] ItemProperties { get; set; }
+        public List<DesignerDataPropertyInfo> ItemProperties { get; set; }
```

Comment:
I believe you should also be able to just do 

`public List<DesignerDataPropertyInfo> ItemProperties { get; set; } = new List<DesignerDataPropertyInfo>();` 

as of a recent-ish C# update.